scripts/whitelist-file.ts

```diff
@@ -0,0 +1,120 @@
+// Simple upload 500 addresses to whitelist using whitelist_addresses.csv.
+// Accepts cosmos, stars addresses.
+// If you run into an error with `member_limit`, run `yarn whitelist --increase-member-limit`
+
+import { ExecuteMsg } from '@stargazezone/types/contracts/whitelist/execute_msg';
+import {
+  coins,
+  DirectSecp256k1HdWallet,
+  GasPrice,
+  MsgExecuteContractEncodeObject,
+  SigningCosmWasmClient,
+} from 'cosmwasm';
+import { toStars } from '../src/utils';
+import inquirer from 'inquirer';
+import { getClient } from '../src/client';
+import { MsgExecuteContract } from 'cosmjs-types/cosmwasm/wasm/v1/tx';
+import { toUtf8 } from '@cosmjs/encoding';
+import * as fs from 'fs';
+import * as path from 'path';
+import { parse } from 'csv-parse';
+
+const config = require('../config');
+const WHITELIST_CREATION_FEE = coins('100000000', 'ustars');
+const MSG_ADD_ADDR_LIMIT = 500;
+
+const gasPrice = GasPrice.fromString('0ustars');
+
+async function addFile() {
+  const wallet = await DirectSecp256k1HdWallet.fromMnemonic(config.mnemonic, {
+    prefix: 'stars',
+  });
+
+  type Whitelist = {
+    address: string;
+  };
+  const __dirname = process.cwd();
+  const csvFilePath = path.resolve(__dirname, './whitelist_addresses.csv');
+  const headers = ['address'];
+  const fileContent = fs.readFileSync(csvFilePath, { encoding: 'utf-8' });
+  let addrs: Array<string> = [];
+  const client = await getClient();
+  if (!config.minter) {
+    throw Error(
+      '"minter" must be set to a minter contract address in config.js'
+    );
+  }
+  await parse(
+    fileContent,
+    {
+      delimiter: ',',
+      columns: headers,
+    },
+    async (error, fileContents: Whitelist[]) => {
+      if (error) {
+        console.error(error);
+      }
+      fileContents.map((fileContents) => addrs.push(fileContents.address));
+      console.log(addrs);
+
+      let validatedAddrs: Array<string> = [];
+      addrs.forEach((addr) => {
+        validatedAddrs.push(toStars(addr));
+      });
+      let uniqueValidatedAddrs = [...new Set(validatedAddrs)];
```

Comment:
What about sorting them?

```suggestion
      const uniqueValidatedAddrs = [...new Set(validatedAddrs)].sort();
```

Reply:
Inside the set (`new Set(validatedAddrs)`), the sorting does not matter. But using `[...mySet]` you convert the set back to an array. Now the question is: should `uniqueValidatedAddrs` be sorted? My impression is that for debugging purposes of

```ts
const msg: ExecuteMsg = {
        add_members: { to_add: uniqueValidatedAddrs },
      };
```

is is nice to always receive the addresses sorted alphabetically.

I double checked this and it turns out, in JavaScript a `Set` [always preserves the insertion order](https://stackoverflow.com/a/33089750/2013738). So sorting the array might be unnecessary or even unwanted if you want to preserve the original order.